File: core/model/src/commonMain/kotlin/io/github/droidkaigi/confsched/model/MultiLangText.kt
package io.github.droidkaigi.confsched.model

public data class MultiLangText(
    val jaTitle: String,
    val enTitle: String,
) {
    val currentLangTitle: String get() = getByLang(defaultLang())

    public fun getByLang(lang: Lang): String {
        return if (lang == Lang.JAPANESE) {
            jaTitle
        } else {
            enTitle
        }
    }

    public companion object
}

File: app-ios/Sources/Model/Entity/MultiLangText.swift
import Foundation

public struct MultiLangText {
    public let currentLangTitle: String
    public let enTitle: String
    public let jaTitle: String
    
    public init(currentLangTitle: String, enTitle: String, jaTitle: String) {
        self.currentLangTitle = currentLangTitle
        self.enTitle = enTitle
        self.jaTitle = jaTitle
    }
}
